### PR TITLE
feat(cli): add failure handling policy option

### PR DESCRIPTION
**Pull request by yitingb**
**Description of changes:**
Add failure handling policy to both client side and server side
**Why is this change necessary:**

**How was this change tested:**
Unit test
**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java [hunk 1]
```diff
@@ -125,6 +127,7 @@ public DeploymentCommand(NucleusAdapterIpc nucleusAdapterIpc) {
         Optional<Path> artifactDirPath = deTilde(artifactDir);
         createLocalDeploymentRequest.setArtifactsDirectoryPath(artifactDirPath.isPresent() ?
                 artifactDirPath.get().toString() : null);
+        createLocalDeploymentRequest.setFailureHandlingPolicy(failureHandlingPolicy == null ? FailureHandlingPolicy.ROLLBACK : failureHandlingPolicy);
```
Comment (MikeDombo) [inline client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java:130]: setting the default to rollback is wrong. The default should be the _current_ behavior, which is to not rollback.

File: client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java [hunk 2]
```diff
@@ -55,7 +56,8 @@ public DeploymentCommand(NucleusAdapterIpc nucleusAdapterIpc) {
      @CommandLine.Option(names = {"-a", "--artifactDir"}, paramLabel = "Artifacts directory") String artifactDir,
      @CommandLine.Option(names = {"--runWith"}, paramLabel = "Component user and/or group") Map<String, String> runWithOptions,
      @CommandLine.Option(names = {"--systemLimits"}, paramLabel = "Component system resource limits") String systemLimits,
-     @CommandLine.Option(names = {"-c", "--update-config"}, paramLabel = "Component configuration") String configUpdate)
+     @CommandLine.Option(names = {"-c", "--update-config"}, paramLabel = "Component configuration") String configUpdate,
+     @CommandLine.Option(names = {"-p", "--failure-policy"}, paramLabel = "Component configuration") FailureHandlingPolicy failureHandlingPolicy)
```
Comment (junfuchen99) [inline client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java:60]: need to fix `paramLabel`

Comment (junfuchen99) [reply client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java:60]: is this input of FailureHandlingPolicy case sensitive? Should we consider taking it in as a String as handle it ourselves?

File: client/src/test/java/com/aws/greengrass/cli/commands/DeploymentCommandTest.java [hunk 3]
```diff
@@ -174,6 +176,25 @@ void GIVEN_WHEN_components_runwith_provided_THEN_request_contains_the_info() thr
...
+        CreateLocalDeploymentRequest request = new CreateLocalDeploymentRequest();
+        request.setFailureHandlingPolicy(FAILURE_HANDLING_POLICY);
+        verify(nucleusAdapteripc).createLocalDeployment(request);
+        assertThat(exitCode, is(0));
+    }
+
+    @Test
+    void GIVEN_WHEN_failure_handling_policy_provided_with_invalid_words_THEN_invalid_request_is_returned() {
+        int exitCode =
+                runCommandLine("deployment", "create", "-p", "Hello World");
```
Comment (junfuchen99) [inline client/src/test/java/com/aws/greengrass/cli/commands/DeploymentCommandTest.java:192]: What error message do customers see today?

**Comment by junfuchen99** [comment]
Please also update the help commands and CLI_messages.properties

File: client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java (hunk 2 above)
Comment (indougnito) [reply client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java:60]: --failure-handling-policy to be consistent with cloud?

File: client/src/test/java/com/aws/greengrass/cli/commands/DeploymentCommandTest.java (hunk 3 above)
Comment (yitingb) [reply client/src/test/java/com/aws/greengrass/cli/commands/DeploymentCommandTest.java:192]: Invalid value for option '--failure-handling-policy': expected one of [ROLLBACK, DO_NOTHING] (case-sensitive) but was 'test'

**Review by junfuchen99 — DISMISSED** [review]

**Review by shaguptashaikh — APPROVED** [review]

**Review by junfuchen99 — APPROVED** [review]

**Pull request merged**